### DISCUSSION TITLE
Invalid Association in jest testing

### DISCUSSION THREAD

**Discussion by junkrun** [sequelize/sequelize#15985] [Help & Questions] [answered]
Hi. I have a next models and associations
```
export const initClass = async (sequelize: Sequelize) => {
  await Class.init({
    id: {
      type: DataTypes.INTEGER,
      autoIncrement: true,
      primaryKey: true
    },
    name: {
      type: DataTypes.STRING,
      unique: true,
      allowNull: false
    },
    createdAt: {
      type: DataTypes.DATE,
      allowNull: false
    },
  },
    {
      sequelize,
      tableName: 'class'
    }
  );
};
```

```
export const initType = async (sequelize: Sequelize) => {
  await Types.init({
    id: {
      type: DataTypes.INTEGER,
      autoIncrement: true,
      primaryKey: true
    },
    name: {
      type: DataTypes.STRING,
      unique: true,
      allowNull: false
    },
    id_class: {
      type: DataTypes.INTEGER,
      allowNull: false
    },
    createdAt: {
      type: DataTypes.DATE,
      allowNull: false
    },
    updatedAt: {
      type: DataTypes.DATE,
      allowNull: false
    },
  },
    {
      sequelize,
      tableName: 'types'
    }
  );
  Class.hasMany(Types, {
    foreignKey: 'id_class',
    sourceKey: 'id',
    as: 'class',
    onDelete: 'CASCADE',
    onUpdate: 'CASCADE'
  });
  Types.belongsTo(Class, {
    foreignKey: 'id_class',
    targetKey: 'id',
    as: 'class',
    onDelete: 'CASCADE',
    onUpdate: 'CASCADE'
  });
};
```
And I try get data by next query
`const data = await TypesModel.getList({ include: [ 'class' ] })`

When I try It in my app i get success resolve. But when i run above code in jest testing It return error with next error

original: error: column class.id_class does not exist
`sql: 'SELECT "Types"."id", "Types"."name", "Types"."logo", "Types"."identify", "Types"."id_class", "Types"."createdAt", "Types"."updatedAt", "class"."id" AS "class.id", "class"."name" AS "class.name", "class"."identify" AS "class.identify", "class"."createdAt" AS "class.createdAt", "class"."updatedAt" AS "class.updatedAt" FROM "types" AS "Types" INNER JOIN "class" AS "class" ON "Types"."id" = "class"."id_class";',`

I noticed that the substitution of values in the resulting sql query is violated **Types"."id" = "class"."id_class**
I use v6.31.0 version
DB: postgresql

**Exiption is appeared only in tests**

**Comment by ephys** [comment, marked as answer]
`id_class` being used on `class` instead of `types` looks like a bug, but one I can't reproduce. If you can replicate the bug in this repository https://github.com/sequelize/sequelize-sscce, feel free to send it to us so we can find the bug
